tests/Doctrine/Tests/ORM/Functional/Ticket/DDC3775Test.php

```diff
@@ -0,0 +1,376 @@
+
+        $pageMember = new DDC3775PageMember();
+        $pageMember->setUser($userTwo);
+        $pageMember->setPage($pageThree);
+        $manager->persist($pageMember);
+
+        // PageRelations
+        $pageRelation = new DDC3775PageRelation();
+        $pageRelation->setPageCreator($page);
+        $pageRelation->setPageReferenced($pageTwo);
+        $manager->persist($pageRelation);
+
+        $pageRelation = new DDC3775PageRelation();
+        $pageRelation->setPageCreator($pageTwo);
+        $pageRelation->setPageReferenced($pageThree);
+        $manager->persist($pageRelation);
+
+        $pageRelation = new DDC3775PageRelation();
+        $pageRelation->setPageCreator($page);
+        $pageRelation->setPageReferenced($pageThree);
+        $manager->persist($pageRelation);
+
+        $manager->flush();
+        $manager->clear();
+    }
+
+    public function testRelationsPagesShouldBeProxyIfNotPreviouslyFetched()
+    {
+        // After setupDoctrine, pages in $pageIdsToFetch have been added to the identitymap.
+        $pageIdsToFetch = [1];
+        $this->setupDoctrineCaches($pageIdsToFetch);
+
+        $qb = $this->_em->getRepository(__NAMESPACE__ . '\DDC3775PageRelation')->createQueryBuilder('r')
+            ->select('r');
+
+        $relations = $qb->getQuery()->getResult();
+
+        foreach ($relations as $relation) {
+            // This page was not fetched before so it should be a proxy
+            if (false === in_array($relation->getPageCreator()->getId(), $pageIdsToFetch)) {
+                $this->assertInstanceOf('Doctrine\ORM\Proxy\Proxy', $relation->getPageCreator());
+            } elseif (false === in_array($relation->getPageReferenced()->getId(), $pageIdsToFetch)) {
+                $this->assertInstanceOf('Doctrine\ORM\Proxy\Proxy', $relation->getPageReferenced());
+            }
+        }
+    }
+
+
+    public function testRelationsPagesShouldHaveNonNullTitleBecauseItIsNonNullable() {
+        // After setupDoctrine, pages in $pageIdsToFetch have been added to the identitymap.
+        $pageIdsToFetch = [1];
+        $this->setupDoctrineCaches($pageIdsToFetch);
+
+        $qb = $this->_em->getRepository(__NAMESPACE__ . '\DDC3775PageRelation')->createQueryBuilder('r')
+            ->select('r, p1, p2')
+            ->leftJoin('r.pageCreator', 'p1')
+            ->leftJoin('r.pageReferenced', 'p2');
+
+        $relations = $qb->getQuery()->getResult();
+
+        foreach ($relations as $relation) {
+            $this->assertNotNull($relation->getPageCreator()->getTitle());
+            $this->assertNotNull($relation->getPageReferenced()->getTitle());
+        }
+    }
+
+    protected function setupDoctrineCaches($preFetchPages = [1]) {
+        $qb = $this->_em->getRepository(__NAMESPACE__ . '\DDC3775Page')->createQueryBuilder('p')
+            ->select('p')
+            ->where('p.id IN (:ids)')
+            ->setParameter('ids', $preFetchPages);
+
+        $allPages = $qb->getQuery()->getResult();
```

Comment:
If I understand it correctly, you expect the `DDC3775Page` objects being loaded here to be proxy instances? They are real instances, and in this case you are only loading page `1` into the `UnitOfWork`.

Here's in sequence what happens:

 - first query - `DDC3775Page#1` is loaded in the UoW
 - second query - `DDC3775Page#1` is re-hydrated (id only), `DDC3775Page#2` is loaded (id only, other properties set to `null`)
 - second query - `DDC3775PageRelation#1`, `DDC3775PageRelation#2`, `DDC3775PageRelation#3` are loaded in memory (next query) due to the `pageMembers` fetch join

At the end of this operation, there are no proxies in memory. What were you expecting here, and why?